server/example/workflow-modelserver-example/src/main/java/com/eclipsesource/glsp/example/modelserver/workflow/model/WorkflowModelServerModelFactory.java

```diff
@@ -15,23 +15,153 @@
  ******************************************************************************/
 package com.eclipsesource.glsp.example.modelserver.workflow.model;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
 import com.eclipsesource.glsp.api.action.kind.RequestModelAction;
 import com.eclipsesource.glsp.api.factory.ModelFactory;
 import com.eclipsesource.glsp.api.model.GraphicalModelState;
 import com.eclipsesource.glsp.api.utils.ClientOptions;
+import com.eclipsesource.glsp.example.modelserver.workflow.wfnotation.DiagramElement;
+import com.eclipsesource.glsp.example.modelserver.workflow.wfnotation.Edge;
+import com.eclipsesource.glsp.example.modelserver.workflow.wfnotation.Shape;
+import com.eclipsesource.glsp.example.workflow.utils.WorkflowBuilder.ActivityNodeBuilder;
+import com.eclipsesource.glsp.example.workflow.utils.WorkflowBuilder.EdgeBuilder;
+import com.eclipsesource.glsp.example.workflow.utils.WorkflowBuilder.TaskNodeBuilder;
+import com.eclipsesource.glsp.example.workflow.utils.WorkflowBuilder.WeightedEdgeBuilder;
+import com.eclipsesource.glsp.example.workflow.wfgraph.ActivityNode;
+import com.eclipsesource.glsp.example.workflow.wfgraph.TaskNode;
+import com.eclipsesource.glsp.example.workflow.wfgraph.WeightedEdge;
+import com.eclipsesource.glsp.graph.DefaultTypes;
+import com.eclipsesource.glsp.graph.GEdge;
 import com.eclipsesource.glsp.graph.GModelRoot;
+import com.eclipsesource.glsp.graph.GNode;
 import com.eclipsesource.glsp.graph.GraphFactory;
+import com.eclipsesource.modelserver.coffee.model.coffee.Flow;
+import com.eclipsesource.modelserver.coffee.model.coffee.Machine;
+import com.eclipsesource.modelserver.coffee.model.coffee.Node;
+import com.eclipsesource.modelserver.coffee.model.coffee.Task;
+import com.eclipsesource.modelserver.coffee.model.coffee.WeightedFlow;
+import com.eclipsesource.modelserver.coffee.model.coffee.Workflow;
 
 public class WorkflowModelServerModelFactory implements ModelFactory {
+	private static final String ROOT_ID = "sprotty";
 
 	@Override
 	public GModelRoot loadModel(RequestModelAction action, GraphicalModelState modelState) {
-		WorkflowModelServerAccess modelAccess = new WorkflowModelServerAccess(
-				action.getOptions().get(ClientOptions.SOURCE_URI));
+		// 1. Load models and create workflow facade
+		String sourceURI = action.getOptions().get(ClientOptions.SOURCE_URI);
+		WorkflowModelServerAccess modelAccess = new WorkflowModelServerAccess(sourceURI);
 		WorkflowFacade workflowFacade = modelAccess.getWorkflowFacade();
+		if(workflowFacade == null) {
+			return createEmptyRoot();
+		}
+		
+		// 2. Check if request model action can be fulfilled by checking which workflow to load
+		// TODO: Add parameter to RequestModelAction to determine which workflow to use
```

Comment:
I'm wondering whether we should rather put the workflow to use into the client options.

Reply:
Yes, you are right, that was badly phrased, we should check the options of course. I added option handling in a follow-up commit.